photon-core/src/main/java/org/photonvision/common/util/math/MathUtils.java

```diff
@@ -201,4 +206,35 @@ public static void rotationToOpencvRvec(Rotation3d rotation, Mat rvecOutput) {
         var axis = rotation.getAxis().times(angle);
         rvecOutput.put(0, 0, axis.getData());
     }
+
+    /**
+     * Orthogonalize an input matrix using a QR decomposition. QR decompositions decompose a
+     * rectangular matrix 'A' such that 'A=QR'. The output matrix Q is the orthogonal matrix. R is
+     * used to ensure we don't flip signs.
```

Comment:
R is an upper triangular matrix. It's not there to ensure you don't flip signs. Ideally, R should have all positive diagonal elements, but sometimes implementations of the QR decomp don't do that. The negative sign fix is what I'd consider an implementation detail.

I'd mention that this function gives you the closest orthogonal matrix to the input.

Reply:
@calcmogul is that wording better?